.circleci/config.yml

```diff
@@ -1,22 +1,66 @@
-version: 2
-jobs:
-  build:
+version: 2.1
+
+executors:
+  node:
     docker:
       - image: circleci/node:10.15
+
+commands:
+  checkout-and-dependencies:
+    description: "Checkout and install dependencies, managing a cache"
     steps:
       - checkout
       - restore_cache:
           keys:
-          - v1-dependencies-{{ checksum "package.json" }}-{{ checksum "yarn.lock" }}
-          - v1-dependencies-{{ checksum "package.json" }}
-          - v1-dependencies-
-      - run: yarn install
+            - v1-dependencies-{{ checksum "package.json" }}-{{ checksum "yarn.lock" }}
+            - v1-dependencies-{{ checksum "package.json" }}
+            - v1-dependencies-
+      # With --frozen-lockfile, the installation will fail if the lockfile is
+      # outdated compared to package.json.
+      - run: yarn install --frozen-lockfile
       - save_cache:
           paths:
+            - .git
             - node_modules
           key: v1-dependencies-{{ checksum "package.json" }}-{{ checksum "yarn.lock" }}
+
+workflows:
```

Comment:
Note: compared to how I did it in the profile UI, the workflows come before the jobs here. This has been suggested by a friend, and I thought it's indeed clearer this way. I think I'll eventually do it in the profile UI too.